akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -43,12 +43,10 @@ private[akkastream] class DocumentStage[T](
       @inline private def tailable = cursor.wrappee.tailable
 
       private var request: () => Future[Option[Response]] = { () =>
+        // Call onFirst immediately to make sure only one request is sent.
         cursor.makeRequest(maxDocs).andThen {
-          case Success(r) if (
-            tailable && r.reply.numberReturned > 0
-          ) => onFirst()
-
-          case Success(_) if (!tailable) => onFirst()
+          case Success(_) =>
```

Comment:
But I think this is the point. If don't update the request function after the first query, it will keep querying the database. If no results return for a long time, the remote mongodb server will be out of memery eventually.